05_concurrency/homework/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type Item struct {
+	Object         interface{}
+	timeOfCreation time.Time
+	TTL            time.Duration
+}
+
+func (item Item) IsExpired() bool {
+	return item.timeOfCreation.Add(item.TTL).Unix() <= time.Now().Unix()
+}
+
+func (item *Item) UpdateTime() {
+	item.timeOfCreation = time.Now()
+}
+
+type InMemoryCache struct {
```

Comment:
It would be fine to have constructor so anyone can import and use the type.
Otherwise, you could add nil check for the internal map on Set or make it exported to let me initialize the type manually.